numba_cuda/numba/cuda/codegen.py

```diff
@@ -298,6 +349,9 @@ def add_linking_library(self, library):
         self._raise_if_finalized()
 
         self._linking_libraries.add(library)
+        self._linking_files.update(library._linking_files)
+        self._setup_functions.extend(library._setup_functions)
+        self._teardown_functions.extend(library._teardown_functions)
```

Comment:
I feel like this is somewhat underrated in the PR description. Code libraries from one nested dispatcher is added into another code library that belongs to top level dispatcher via `add_linking_library` call. Without this patch, much information from lower level library is lost.

Reply:
I think you're saying that the idea is that whenever we link to a code library, we now make sure to capture all the information from the linked code library - is my understanding correct?

The general idea of this PR was that we should always be keeping track of the required information for linking correctly, and not require any special treatment in the dispatcher. This does make me think that as a follow on, there should be an improvement to the `cuda.compile()` API to allow the linking dependencies to be determined - otherwise you can compile code that you have to know how to link. (I don't want to try to tackle this straight away, because this PR addresses a concrete deficiency that exists right now).